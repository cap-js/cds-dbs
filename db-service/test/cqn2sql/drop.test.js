'use strict'
const cds = require('@sap/cds/lib')
const cqn2sql = require('../../lib/cqn2sql')

beforeAll(async () => {
  cds.model = await cds.load(__dirname + '/testModel').then(cds.linked)
})
describe('drop', () => {
  test('test drop table with string entity', () => {
    const { sql } = cqn2sql({
      DROP: {
        entity: 'Foo',
      },
    })
    expect({ sql }).toMatchSnapshot()
  })
  test('test drop table with string table', () => {
    const { sql } = cqn2sql({
      DROP: {
        entity: 'Foo',
        table: 'Foo',
      },
    })
    expect({ sql }).toMatchSnapshot()
  })

  xtest('test drop table with ref', () => {
    const { sql } = cqn2sql({
      DROP: {
        table: { ref: ['Foo'] },
      },
    })
    expect({ sql }).toMatchSnapshot()
    // Cannot destructure property 'ref' of 'from' as it is undefined.
  })

  xtest('test drop table with ref', () => {
    const { sql } = cqn2sql({
      DROP: {
        view: { ref: ['Foo'] },
      },
    })
    expect({ sql }).toMatchSnapshot()
    // Cannot destructure property 'ref' of 'from' as it is undefined.
  })

  xtest('test drop view with string view', () => {
    const { sql } = cqn2sql({
      DROP: {
        view: 'Foo',
      },
    })
    expect({ sql }).toMatchSnapshot()
    //TypeError: Cannot destructure property 'ref' of 'from' as it is undefined.
  })

  xtest('test drop table with CQN entity', () => {
    const { sql } = cqn2sql({
      DROP: {
        entity: { ref: 'Books' },
      },
    })
    expect({ sql }).toMatchSnapshot()
    //"B" not found in the definitions of your model
  })
})
