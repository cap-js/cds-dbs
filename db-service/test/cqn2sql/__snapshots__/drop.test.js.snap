// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drop test drop table with string entity 1`] = `
{
  "sql": "DROP TABLE IF EXISTS Foo",
}
`;

exports[`drop test drop table with string table 1`] = `
{
  "sql": "DROP TABLE IF EXISTS Foo",
}
`;
